pkg/operators/ebpf/ebpf.go

```diff
@@ -679,35 +679,36 @@ func (i *ebpfInstance) Start(gadgetCtx operators.GadgetContext) error {
 	}
 	i.collection = collection
 
-	if i.bpfOperator.gadgetObjs != nil {
-		// collect program IDs and map IDs for this gadget
-		gadgetObjs := gadgetObjs{}
+	// collect program IDs and map IDs for this gadget
+	gadgetObjs := gadgetObjects{}
 
-		for _, p := range i.collection.Programs {
-			info, err := p.Info()
-			if err != nil {
-				i.logger.Warnf("stats for this gadget won't be available: getting program info: %v", err)
-				continue
-			}
-
-			id, _ := info.ID()
-			gadgetObjs.programIDs = append(gadgetObjs.programIDs, id)
+	for _, p := range i.collection.Programs {
+		info, err := p.Info()
+		if err != nil {
+			i.logger.Warnf("stats for this gadget won't be available: getting program info: %v", err)
+			continue
 		}
 
-		for _, m := range i.collection.Maps {
-			info, err := m.Info()
-			if err != nil {
-				i.logger.Warnf("stats for this gadget won't be available: getting map info: %v", err)
-				continue
-			}
+		id, _ := info.ID()
+		gadgetObjs.programIDs = append(gadgetObjs.programIDs, id)
+	}
 
-			id, _ := info.ID()
-			gadgetObjs.mapIDs = append(gadgetObjs.mapIDs, id)
+	for _, m := range i.collection.Maps {
+		info, err := m.Info()
+		if err != nil {
+			i.logger.Warnf("stats for this gadget won't be available: getting map info: %v", err)
+			continue
 		}
-		i.bpfOperator.mu.Lock()
-		i.bpfOperator.gadgetObjs[gadgetCtx] = gadgetObjs
-		i.bpfOperator.mu.Unlock()
+
+		id, _ := info.ID()
+		gadgetObjs.mapIDs = append(gadgetObjs.mapIDs, id)
+	}
+	i.bpfOperator.mu.Lock()
+	if i.bpfOperator.gadgetObjs == nil {
+		i.bpfOperator.gadgetObjs = make(map[operators.GadgetContext]gadgetObjects)
 	}
```

Comment:
Should we move this to the init code where the other maps are initialized as well?

Reply:
If I move this to `func (o *ebpfOperator) Init(params *params.Params)` then a lot of tests and applications using the Go API will crash as they don't call that function. I could move it to `InstantiateImageOperator()` but I don't see any different with having it on `Start()`.

Reply:
Makes sense - nevermind, then.